### ISSUE TITLE
OnDeviceTraining

### ISSUE THREAD

**Issue by LeoBuron** [es-ude/elastic-ai.runtime.enV5#434]
- [x] Arithmetic
  - inplace
  - tensor with element
  - tensor with tensor
  
- [x] Renaming linearQ -> asym
- [x] implement SymInt32 and Sym Quantization
- [x] refactor linearType to FLOATLINEAR, SYMLINEAR und ASYMLINEAR
      - SYMINT32 macht keinen Sinn als layer type, ist nur Schnittstelle
      - Asym nutzt eine nach außen nicht sichtbare Implementierung von linearForwardSymInt32
- [x] SymInt32 Tensor Conversion 
      1. implement asym -> symInt32 / symInt32 -> asym)
      2. refactor linearToFloat: asym -> symInt32 -> float/int
- [x] SymInt32 Math Lib
      - matmul, mul, div happens without casting to 
- [x] Remove float from linear forward output
- [x] Remove matmulTensors
- [x] Tests for SymInt32 Math Lib 

- [x] Layers
  1. Linear (ASYM via SYMINT32 Abstraktion, SYM | BIAS IS ALWAYS INT32, GRADIENTS ARE NOT)
  2. ReLU
- [x] MSE
  1.  MSEBackwardFloat
  2. MSEBackwardAsym
- [x] SGD
- [x] Sequential
  1. Alles Float
  2. Alles AsymQ
- [x] CrossEntropy, Softmax
    - [x] Float
    - [x] Asym 
- [x] Arithmetic Logarithmus
- [x] CSV
- [x] move to dedicated repo
- [ ] Gradient checkpointing
- [ ] Improve memory usage in calcGrads
- [ ] 1D Convolution (kein Stride, Dilation, Padding)
- [ ] Convolution mit Stride, Dilation, Padding
- [ ] linearQ forward -> float backward

**Comment by DavidFederl** [comment]
@LeoBuron can this issue be closed?

**Comment by LeoBuron** [comment]
Yes
